dbt_semantic_interfaces/validations/saved_query.py

```diff
@@ -136,9 +138,44 @@ def _check_where(saved_query: SavedQuery) -> Sequence[ValidationIssue]:
                         },
                     )
                 )
+            else:
+                issues += SavedQueryRule._check_where_timespine(saved_query, custom_granularity_names)
 
         return issues
 
+    def _check_where_timespine(
```

Comment:
For reviewers: This should look like the code in [metrics.py](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/98b5af50ecf4c8d9730b3b24476cbec63e0aaad3/dbt_semantic_interfaces/validations/metrics.py#L251), adjusted a bit for the calling context

Reply:
Related to this - can we dedupe this logic? There is a ton reused between here and `metrics.py`. I'm thinking we should move `WhereFiltersAreParseable` to its own file and move most of this logic to that class, maybe in a method called `_validate_saved_query()`. I think you should be able to use `_validate_time_granularity_names()` for both metrics and saved queries and that should contain all the duplicate logic, you just might need to update the context param to be more generic (if possible).

Reply:
Also, if you do that, then it might make sense to take the manifest as a param for `WhereFiltersAreParseable` on `__init__`, and you can parse out the custom granularity names and store them there. That would also mean anything that uses the custom granularity names won't be a `staticmethod`, but at least the logic that parses out custom granularity names would be encapsulated in a place that makes sense.

Reply:
I'll see what I can do. :-)

Reply:
I'm going to resolve this thread, but please look at the commit message for the most recent commit (I'll probably copy-paste it as a comment on the PR, too, just to be safe) because I structured things a little differently.